litellm/litellm_core_utils/litellm_logging.py

```diff
@@ -2166,3 +2181,156 @@ def use_custom_pricing_for_model(litellm_params: Optional[dict]) -> bool:
             if k in SPECIAL_MODEL_INFO_PARAMS:
                 return True
     return False
+
+
+def is_valid_sha256_hash(value: str) -> bool:
+    # Check if the value is a valid SHA-256 hash (64 hexadecimal characters)
+    return bool(re.fullmatch(r"[a-fA-F0-9]{64}", value))
+
+
+def get_standard_logging_object_payload(
+    kwargs: Optional[dict],
+    init_response_obj: Any,
+    start_time: dt_object,
+    end_time: dt_object,
+) -> Optional[StandardLoggingPayload]:
+    try:
+        if kwargs is None:
+            kwargs = {}
+
+        hidden_params: Optional[dict] = None
+        if init_response_obj is None:
+            response_obj = {}
+        elif isinstance(init_response_obj, BaseModel):
+            response_obj = init_response_obj.model_dump()
+            hidden_params = getattr(init_response_obj, "_hidden_params", None)
+        else:
+            response_obj = {}
+        # standardize this function to be used across, s3, dynamoDB, langfuse logging
+        litellm_params = kwargs.get("litellm_params", {})
+        proxy_server_request = litellm_params.get("proxy_server_request") or {}
+        end_user_id = proxy_server_request.get("body", {}).get("user", None)
+        metadata = (
+            litellm_params.get("metadata", {}) or {}
+        )  # if litellm_params['metadata'] == None
+        completion_start_time = kwargs.get("completion_start_time", end_time)
+        call_type = kwargs.get("call_type")
+        cache_hit = kwargs.get("cache_hit", False)
+        usage = response_obj.get("usage", None) or {}
+        if type(usage) == litellm.Usage:
+            usage = dict(usage)
+        id = response_obj.get("id", kwargs.get("litellm_call_id"))
+
+        _model_id = metadata.get("model_info", {}).get("id", "")
+        _model_group = metadata.get("model_group", "")
+
+        request_tags = (
+            metadata.get("tags", [])
+            if isinstance(metadata.get("tags", []), list)
+            else []
+        )
+
+        # cleanup timestamps
+        if isinstance(start_time, datetime.datetime):
+            start_time_float = start_time.timestamp()
+        elif isinstance(start_time, float):
+            start_time_float = start_time
+        if isinstance(end_time, datetime.datetime):
+            end_time_float = end_time.timestamp()
+        elif isinstance(end_time, float):
+            end_time_float = end_time
+        if isinstance(completion_start_time, datetime.datetime):
+            completion_start_time_float = completion_start_time.timestamp()
+        elif isinstance(completion_start_time, float):
+            completion_start_time_float = completion_start_time
+        # clean up litellm hidden params
+        clean_hidden_params = StandardLoggingHiddenParams(
+            model_id=None,
+            cache_key=None,
+            api_base=None,
+            response_cost=None,
```

Comment:
Tiny comment on naming: I think `response_cost` is slightly ambiguous in whether it's the cost for the entire request (prompt tokens + completion tokens), or if it's the cost of the completion only. I'm assuming it's the former, but there is perhaps some room for ambiguity.

Also, how does this play together with `cache_hit`? I mean, when the response comes from the cache, actual cost is zero. But will this show a zero or what the price would've been? The latter is useful information, so I hope it is included. Calling it `price` could be logical in that case (and then "cost" would be zero when cached and `price` otherwise).